core/src/main/java/org/apache/brooklyn/entity/group/DynamicClusterImpl.java

```diff
@@ -151,12 +151,14 @@ public ZoneFailureDetector apply(final String input) {
      */
     protected final Object mutex = new Object[0];
 
-    private static final Function<Collection<Entity>, Entity> defaultRemovalStrategy = new Function<Collection<Entity>, Entity>() {
-        @Override public Entity apply(Collection<Entity> contenders) {
```

Comment:
Rule of thumb is that anything that goes into a config or sensor (or is nested in such an object) will get persisted. Looking at the code this is not the case, but can never know what an extending class might be doing, so better be on the safe side.